test/format/util.c

```diff
@@ -572,25 +572,45 @@ checkpoint(void *arg)
 }
 
 /*
- * timestamp --
- *     Periodically update the oldest timestamp.
+ * timestamp_once --
+ *     Update the timestamp once.
  */
-WT_THREAD_RET
-timestamp(void *arg)
+void
+timestamp_once(void)
 {
     WT_CONNECTION *conn;
     WT_DECL_RET;
-    WT_SESSION *session;
     char buf[WT_TS_HEX_STRING_SIZE + 64];
-    bool done;
 
-    (void)(arg);
     conn = g.wts_conn;
 
-    testutil_check(conn->open_session(conn, NULL, NULL, &session));
-
     testutil_check(__wt_snprintf(buf, sizeof(buf), "%s", "oldest_timestamp="));
 
+    /*
+     * Lock out transaction timestamp operations. The lock acts as a barrier ensuring we've checked
+     * if the workers have finished, we don't want that line reordered.
+     */
+    testutil_check(pthread_rwlock_wrlock(&g.ts_lock));
+
+    ret = conn->query_timestamp(conn, buf + strlen("oldest_timestamp="), "get=all_committed");
```

Comment:
Should change `all_committed => all_durable`. They alias to the same timestamp but `all_committed` is deprecated and will be removed in the near future (whenever someone gets around to it).
Also it might be nice to give this string a name as I git a bit confused here. Maybe:
```
static const char *oldest_timestamp_str = "oldest_timestamp=";
```

Reply:
Pushed in a792a4e38.